jsk_perception/node_scripts/aws_auto_checkin_app.py

```diff
@@ -50,34 +44,88 @@ def __init__(self):
+            self.COLLECTION_ID = env['CollectionId']
+            self.DYNAMODB_TABLE = env['DynamodbTable']
+            self.MAX_FACES = env['MaxFaces']
+            region_name = env['Region']
+        except KeyError:
+            print('Invalid config file')
+            raise
+
+        aws_credentials_path = rospy.get_param('~aws_credentials_path', 'aws.json')
+        rospy.loginfo("Loading AWS credentials from {}".format(aws_credentials_path))
+        try:
+            with open(aws_credentials_path) as aws_json:
+                aws_credentials = json.load(aws_json)
+        except IOError:
+            rospy.logerr('Cannot open "{}".\n Please put region/aws_access_key_id/aws_secret_access_key to aws.json.'.format(aws_credentials_path))
+            sys.exit(1)
+
+        try:
+            aws_access_key_id = aws_credentials['aws_access_key_id']
+            aws_secret_access_key = aws_credentials['aws_secret_access_key']
         except KeyError:
             print('Invalid config file')
             raise
 
-        self.id_token = self._get_id_token_by_cognito(env['UserName'], env['UserPassword'])
+        self.rekognition = boto3.client(
+            'rekognition',
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            region_name=region_name)
+
+        self.dynamodb =  boto3.resource(
+            'dynamodb',
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            region_name=region_name)
+        self.dynamodb_table = self.dynamodb.Table(self.DYNAMODB_TABLE)
 
         self.use_window = rospy.get_param('~use_window', False)
         rospy.loginfo("Launch image window : {}".format(self.use_window))
 
-        self.name_pub = rospy.Publisher('face_name', FaceArrayStamped, queue_size=1)
+        self.classifier_name = rospy.get_param(
+            "~classifier_name", rospy.get_name())
+        self.target_names = self.get_label_names()
+        self.target_name_to_label_index = {}
+        for i, name in enumerate(self.target_names):
+            self.target_name_to_label_index[name] = i
+
+        self.name_pub = self.advertise('face_name', FaceArrayStamped, queue_size=1)
+        self.pub_rects = self.advertise("~output/rects", RectArray,
+                                        queue_size=1)
+        self.pub_class = self.advertise("~output/class", ClassificationResult,
+                                        queue_size=1)
+
+    def subscribe(self):
         self.image_sub = message_filters.Subscriber('{}/compressed'.format(rospy.resolve_name('image')), CompressedImage)
-        # we wan to use RegionOfInterest, but it message_filters requires
-        # header information, so use CameraInfo
         self.roi_sub = message_filters.Subscriber('face_roi', FaceArrayStamped)
-        self.ts = message_filters.ApproximateTimeSynchronizer([self.image_sub, self.roi_sub], 10, 1, allow_headerless = True)
```

Comment:
please consider using 
```
        self.transport_hint = rospy.get_param(
            namespace + 'image_transport', 'compressed')
```
like
https://github.com/jsk-ros-pkg/coral_usb_ros/blob/2c5ddeb033a094795db927c31685fd7958104f8f/python/coral_usb/node_base.py#L128-L135

so that we can switch compressed image / raw image 
c.f.
http://wiki.ros.org/image_transport/Tutorials/ExaminingImagePublisherSubscriber

Reply:
Thank you. raw image becomes available by 0fdd1d8